lib/google_analytics/analytics_export_service.rb

```diff
@@ -0,0 +1,36 @@
+require 'google/apis/analytics_v3'
+require 'googleauth'
+
+module GoogleAnalytics
+  class AnalyticsExportService
+    include Google::Apis::AnalyticsV3
+    include Google::Auth
+
+    attr_accessor :service, :account_id, :web_property_id, :custom_data_source_id
+
+    def build
+      @service = AnalyticsService.new
+      @service.authorization ||= ServiceAccountCredentials.make_creds(
+        json_key_io: File.open(File.join(Rails.root, 'config', 'analytics_exporter_client_secrets.json')),
+        scope: 'https://www.googleapis.com/auth/analytics.edit')
+      @service
+    end
+
+    def export_bad_links(data)
+      begin
+        file = Tempfile.new('bad_links.csv')
```

Comment:
I *think* if you use the block form of [`Tempfile.create`](https://ruby-doc.org/stdlib-2.3.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-create) it cleans everything up for you, even if there is an exception.  Worth checking.

Reply:
Yes it does!